ssh/tailssh/tailssh.go

```diff
@@ -219,15 +222,12 @@ type conn struct {
 	idH    string
 	connID string // ID that's shared with control
 
-	// anyPasswordIsOkay is whether the client is authorized but has requested
-	// password-based auth to work around their buggy SSH client. When set, we
-	// accept any password in the PasswordHandler.
-	anyPasswordIsOkay bool // set by NoClientAuthCallback
```

Comment:
This is no longer needed since we only specify a passwordCallback [after](https://github.com/tailscale/tailscale/pull/14803/files#diff-3f7d7fbab09c188bb05cc9c2a319a87e7da2b93b5e806a669c5771cdd7d9f908R381) successful Tailscale authentication.